plugins/core-config/hooks/notify_osc.py

```
@@ -16,10 +16,41 @@ def send_notification(title: str, body: str):
+    The text is handed to an ``on run argv`` handler as arguments rather than
+    interpolated into the AppleScript source. Interpolating would break on a
+    title containing a double quote, and would make any notification text
+    executable AppleScript -- the body can come from a hook payload.
+    """
+    script = (
+        "on run argv\n"
+        "display notification (item 1 of argv) with title (item 2 of argv)\n"
+        "end run"
+    )
+    try:
+        proc = subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)